compiler/x/codegen/OMRTreeEvaluator.hpp

```diff
@@ -286,6 +286,15 @@ class OMR_EXTENSIBLE TreeEvaluator: public OMR::TreeEvaluator
    // routines for floating point values that can fit in one GPR
    static TR::Register *floatingPointStoreEvaluator(TR::Node *node, TR::CodeGenerator *cg);
 
+   // SIMD evaluators
+   static TR::Register *BinaryArithmeticEvaluator(TR::Node *node, TR::CodeGenerator *cg);
```

Comment:
No, it is not purely for SIMD. Although it is only used by vadd and vmul, the evaluator is designed to handle generic cases where ILOpCode can be translate to a single SSE/AVX instruction.